### PR TITLE
Add NIST data

### PR DESCRIPTION
**Pull request by s-rathi**
### :pencil: Workflow to download the NIST data

**Type:** :rocket: `feature` 

This workflow will run every Sunday at midnight and download the NIST ionization energies and atomic weights. The data will be saved in a `CSV` file.

### :pushpin: Resources

notebooks, and links to useful references.

### :vertical_traffic_light: Testing

How did you test these changes?

- [ ] Testing pipeline
- [x] Other method (describe)
- [ ] My changes can't be tested (explain why)

### :ballot_box_with_check: Checklist

- [x] I requested two reviewers for this pull request
- [ ] I updated the documentation according to my changes
- [ ] I built the documentation by applying the `build_docs` label

> **Note:** If you are not allowed to perform any of these actions, ping (@) a contributor.

### CODE REVIEW

File: ionization.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+# Remove empty rows
+table_data = [row for row in table_data if len(row) > 1]  
+
+# Convert the table data into a DataFrame
+column=['At. Num', 'Ion Charge', 'El. Name', 'Ground Shells', 'Ground Level', 'Ionization Energy (eV)', 'Uncertainty (eV)', 'x']
+df = pd.DataFrame(table_data[2:], columns=column)
+df = df.drop(df.columns[-1], axis=1)
+
+# Save the DataFrame to a CSV file
+df.to_csv('nist_ionization_energies.csv', index=False)
```
Comment (atharva-2001) [inline ionization.py]: All this in a function perhaps?

File: .github/workflows/ionization_energies.yml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+          
+      - name: Set up Python
+        id: python_setup
+        uses: actions/setup-python@v3
+        
+      - name: Install dependencies
+        id: install_dependencies
+        run: | 
+          pip install -r requirements.txt
+          pip install html5lib
```
Comment (atharva-2001) [inline .github/workflows/ionization_energies.yml]: Why not add this to the requirements.txt?

File: .github/workflows/ionization_energies.yml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        run: |
+          git add .
+          git diff --quiet --exit-code --cached
+
+      - name: Commit files to git
+        id: commit_changes
+        if: ${{ steps.compare.outcome == 'failure' }}
+        run: |
+          git commit -m "nist ionization energies added"
+
```
Comment (atharva-2001) [inline .github/workflows/ionization_energies.yml]: why not do both ionization energies and atomic weights at the same time?

**Comment by atharva-2001** [comment]
Hi Shikha, both of these PRs look very good, I just made a few small comments. Could you please make these changes so  that these PRs can be merged?

File: .github/workflows/ionization_energies.yml (hunk 2 above)
Comment (s-rathi) [reply .github/workflows/ionization_energies.yml]: Ohh! I see its already there in requirements.txt. We can remove `pip install html5lib` from the workflow.

File: .github/workflows/ionization_energies.yml (hunk 3 above)
Comment (s-rathi) [reply .github/workflows/ionization_energies.yml]: That's a good idea, a single workflow for both parameters. I will make the required changes. Thanks.

Comment (atharva-2001) [reply .github/workflows/ionization_energies.yml]: Can you also bring this up in the carsus meeting?

File: atomic_weights.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
+import requests
+import pandas as pd
+from bs4 import BeautifulSoup
+import os
+from carsus.io.nist.weightscomp import download_weightscomp
```
Comment (atharva-2001) [inline atomic_weights.py]: Small thing- can you run isort on both of these files?

File: .github/workflows/nist_data.yml [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+        id: run_atomic_weights_script
+        run: | 
+          python atomic_weights.py
+          
+      - name: Compare changes
+        id: compare
+        continue-on-error: true
+        run: |
+          git add .
+          git rm --cached -f carsus
```
Comment (atharva-2001) [inline .github/workflows/nist_data.yml]: I don't think we should use the -f option, did the `git add abc xyz not work`?

Comment (s-rathi) [reply .github/workflows/nist_data.yml]: We have to use `-f` and do force removal, otherwise it's not removed.

File: atomic_weights.py (hunk 4 above)
Comment (s-rathi) [reply atomic_weights.py:5]: Ohh! I was wrong. It's working. I have modify the code

File: .github/workflows/nist_data.yml (hunk 5 above)
Comment (s-rathi) [reply .github/workflows/nist_data.yml]: Ohh! I was wrong. It's working. I have modified the code

File: atomic_weights.py (hunk 4 above)
Comment (s-rathi) [reply atomic_weights.py]: Sure

**Review by atharva-2001 — APPROVED** [review]

**Review by andrewfullard — APPROVED** [review]

**Pull request merged**